### PR TITLE
Fix almost all -Wall warnings

### LINKED ISSUE #713: Compiler warnings elimination
Starting from the creation of this task, #116 is a subtask of this task.

MSVC:
- [ ] MSBuild
  - [ ] Enable "warnings are errors" mode *(is it possible?)*
- [ ] Compiler
  - [x] Enable conformance mode
  - [x] Enable "warnings are errors" mode
  - [ ] Raise the warning level to 4
    - [x] Level 1
    - [x] Level 2
    - [ ] Level 3
    - [ ] Level 4
- [ ] Linker
  - [ ] Enable "warnings are errors" mode

GCC:
- CMake and `make`
  - [ ] Enable "warnings are errors" mode *(is it possible?)*
- [ ] Compiler
  - [x] Disable permissive mode
  - [ ] *Maybe enable [pedantic mode](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html)*
  - [ ] Enable "warnings are errors" mode
  - [ ] Raise the warning level to 4
    - [ ] Level 1
    - [ ] Level 2
    - [ ] Level 3
    - [ ] Level 4
- [ ] Linker
  - [ ] Enable "warnings are errors" mode

For history:
- [ ] List all warnings elimination commits in this issue

Tools:
PVS-Studio
CppDepend (includes cppcheck, Clang-Tidy, Vera++)
ReSharper C++ (includes Clang and Clang-Tidy check plus it's own analysis)
Visual Assist (has it's own code inspection)

https://habr.com/ru/articles/75123/
https://habr.com/ru/companies/pvs-studio/articles/303664/

### PR DESCRIPTION
**Pull request by Masterkatze**

### CODE REVIEW

File: src/Layers/xrRenderPC_GL/r2_R_sun.cpp [hunk 1]
```diff
@@ -711,7 +719,7 @@ void CRender::render_sun()
         DumbClipper view_clipper;
         glm::mat4 xform = m_LightViewProj;
         view_clipper.frustum.CreateFromMatrix(*(Fmatrix*)glm::value_ptr(ex_full), FRUSTUM_P_ALL);
-        for (int p = 0; p < view_clipper.frustum.p_count; p++)
+        for (std::size_t p = 0; p < view_clipper.frustum.p_count; p++)
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderPC_GL/r2_R_sun.cpp]: Standard for this engine is `size_t`, not `std::size_t`

File: src/Layers/xrRenderPC_GL/r2_R_sun.cpp [hunk 2]
```diff
@@ -300,16 +300,23 @@ struct DumbClipper
             EFC_Visible res = frustum.testAABB(&bb.vMin.x, mask);
             switch (res)
             {
-            case fcvNone: continue;
+            case fcvNone:
+            case fcv_forcedword:
+            {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderPC_GL/r2_R_sun.cpp]: These parentheses are excessive and not needed in this entire diff.
Also, default case and missing cases are allowed to be missing. I believe, this warning should be disabled.

File: src/Layers/xrRenderPC_GL/gl_rendertarget_phase_scene.cpp [hunk 3]
```diff
@@ -19,7 +19,7 @@ void CRenderTarget::phase_scene_prepare()
     if (RImplementation.o.advancedpp &&
         (
             ps_r2_ls_flags.test(R2FLAG_SOFT_PARTICLES | R2FLAG_DOF) ||
-            ps_r_sun_shafts > 0 && fValue >= 0.0001 ||
+            (ps_r_sun_shafts > 0 && fValue >= 0.0001) ||
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderPC_GL/gl_rendertarget_phase_scene.cpp]: This warning should be disabled

File: src/Layers/xrRender/dxUIRender.cpp [hunk 4]
```diff
@@ -187,14 +187,22 @@ void dxUIRender::PushPoint(float x, float y, float z, u32 C, float u, float v)
     switch (m_PointType)
     {
     case pttLIT:
+    {
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/dxUIRender.cpp]: It that's a warning, it should be disabled

File: src/Layers/xrRender/dxFontRender.cpp [hunk 5]
```diff
@@ -88,6 +88,7 @@ void dxFontRender::OnRender(CGameFont& owner)
                 {
                 case CGameFont::alCenter: X -= (iFloor(fSize * 0.5f)) * g_current_font_scale.x; break;
                 case CGameFont::alRight: X -= iFloor(fSize); break;
+                case CGameFont::alLeft: break; // XXX: do something with X here
```
Comment (Xottab-DUTY) [inline src/Layers/xrRender/dxFontRender.cpp]: Do nothing, this warning should be disabled.

File: src/xrCDB/xrCDB.cpp [hunk 6]
```diff
@@ -43,7 +43,9 @@ MODEL::~MODEL()
 
 void MODEL::syncronize_impl() const
 {
-    Log("! WARNING: syncronized CDB::query");
+    #ifdef DEBUG
+        Msg("! WARNING: syncronized CDB::query");
+    #endif
```
Comment (Xottab-DUTY) [inline src/xrCDB/xrCDB.cpp]: It's suboptimal to use `Msg` when formatting is not needed. In this case, the most optimal choice is `Log`.
Plus, formatting.
```suggestion
#ifdef DEBUG
    Log("! WARNING: syncronized CDB::query");
#endif
```

File: src/Layers/xrRenderGL/glSH_RT.cpp [hunk 7]
```diff
@@ -22,14 +22,16 @@ CRT::~CRT()
...
+        return true;
+
     switch (fmt)
     {
     case D3DFMT_D16:
     case D3DFMT_D16_LOCKABLE:
     case D3DFMT_D15S1:
     case D3DFMT_D24X8:
     case D3DFMT_D24S8:
-    case MAKEFOURCC('D', 'F', '2', '4'):
```
Comment (Xottab-DUTY) [inline src/Layers/xrRenderGL/glSH_RT.cpp]: What's the warning that complains about this?

File: Externals/ode/ode/src/collision_quadtreespace.cpp [hunk 8]
```diff
@@ -347,7 +347,7 @@ struct dxQuadTreeSpace : public dxSpace{
 };
 
 dxQuadTreeSpace::dxQuadTreeSpace(dSpaceID _space, dVector3 Center, dVector3 Extents, int Depth) : dxSpace(_space){
-	type = dQuadTreeSpaceClass;
+    type = dClassNumber::dQuadTreeSpaceClass;
```
Comment (Xottab-DUTY) [inline Externals/ode/ode/src/collision_quadtreespace.cpp]: ODE uses tabs everywhere, please don't replace them with spaces.
```suggestion
	type = dClassNumber::dQuadTreeSpaceClass;
```

File: src/xrGame/PHCommander.cpp [hunk 9]
```diff
@@ -236,5 +238,5 @@ void CPHCommander::RemoveCallsDeferred(CPHReqComparerV* comparer)
 void CPHCommander::phys_shell_relcase(CPhysicsShell* sh)
 {
     CPHReqComparerHasShell c(sh);
-    remove_calls_threadsafety(&c);
+    remove_calls(&c);
```
Comment (Xottab-DUTY) [inline src/xrGame/PHCommander.cpp]: That's unrelated to this PR

File: src/xrGame/PHCommander.h [hunk 10]
```diff
@@ -99,7 +99,7 @@ class CPHCommander : public IPHWorldUpdateCallbck
     void clear();
 
 private:
-    virtual void update_step() { update_threadsafety(); }
+    virtual void update_step() { update(); }
```
Comment (Xottab-DUTY) [inline src/xrGame/PHCommander.h]: That's unrelated to this PR

File: src/xrGame/PdaMsg.h [hunk 11]
```diff
@@ -33,12 +33,14 @@ typedef struct tagSPdaMessage
...
 {
-    TALK_CONTACT_DATA() : id(u16(-1)), time(0){};
+    TALK_CONTACT_DATA() = default;
     TALK_CONTACT_DATA(u16 contact_id, ALife::_TIME_ID contact_time) : id(contact_id), time(contact_time){};
-    //время контакта
-    ALife::_TIME_ID time;
+
     // id персонажа с которым говорили
-    u16 id;
+    u16 id{u16(-1)};
```
Comment (Xottab-DUTY) [inline src/xrGame/PdaMsg.h]:
```suggestion
    u16 id{ u16(-1) };
```

File: src/xrUICore/Windows/UIWindow.h [hunk 12]
```diff
@@ -142,34 +142,34 @@ class XRUICORE_API CUIWindow : public CUISimpleWindow
...
 
     // Is user input allowed
-    bool m_bIsEnabled;
+    bool m_bIsEnabled{false};
 
     // Если курсор над окном
-    bool m_bCursorOverWindow;
-    bool m_bCustomDraw;
+    bool m_bCursorOverWindow{false};
+    bool m_bCustomDraw{false};
```
Comment (Xottab-DUTY) [inline src/xrUICore/Windows/UIWindow.h]: Everything should be initialized using `{}`.
```suggestion
    bool m_bCustomDraw{};
```

File: src/Layers/xrRenderGL/glSH_RT.cpp (hunk 7 above)
Comment (Masterkatze) [reply src/Layers/xrRenderGL/glSH_RT.cpp]: `error: case value ‘875710020’ not in enumerated type ‘const D3DFORMAT’ [-Werror=switch]`

**Comment by OPNA2608** [comment]
Errors on my end, x86_64 Linux w/ GCC10 (and 9):
```
[ 10%] Building CXX object src/xrCore/CMakeFiles/xrCore.dir/FS.cpp.o
/build/source/src/xrCore/FS.cpp: In function 'void FileCompress(pcstr, pcstr, void*, size_t)':
/build/source/src/xrCore/FS.cpp:172:11: error: ignoring return value of 'ssize_t write(int, const void*, size_t)' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Werror=unused-result8;;]
  172 |     _write(H, &M, 8);
      |           ^
/build/source/src/xrCore/FS.cpp: In function 'void* FileDecompress(pcstr, pcstr, size_t*)':
/build/source/src/xrCore/FS.cpp:184:10: error: ignoring return value of 'ssize_t read(int, void*, size_t)' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result-Werror=unused-result8;;]
  184 |     _read(H, &F, 8);
      |          ^
cc1plus: all warnings being treated as errors
make[2]: *** [src/xrCore/CMakeFiles/xrCore.dir/build.make:237: src/xrCore/CMakeFiles/xrCore.dir/FS.cpp.o] Error 1
```

**Comment by Masterkatze** [comment]
@OPNA2608 Hi, sorry for such late answer, can you try it now?

**Comment by OPNA2608** [comment]
Doesn't error anymore, but some warnings remain on my end. [Here's the build log](https://github.com/OpenXRay/xray-16/files/8644794/pr-907.log).

**Comment by Masterkatze** [comment]
Случайно закрыл

**Comment by Xottab-DUTY** [comment]
Since you have submitted a major number of changes from this PR in separate PRs, let's close it to reduce the number of pull requests. :)

**Pull request closed without merging**